include/oneapi/mkl/dft/detail/dft_loader.hpp

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#ifndef _ONEMKL_DFT_LOADER_HPP_
+#define _ONEMKL_DFT_LOADER_HPP_
+
+#include <cstdint>
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+#include "oneapi/mkl/detail/export.hpp"
+#include "oneapi/mkl/detail/get_device_id.hpp"
+
+#include "oneapi/mkl/types.hpp"
+#include "oneapi/mkl/dft/types.hpp"
+#include "oneapi/mkl/dft/detail/commit_impl.hpp"
+#include "oneapi/mkl/dft/descriptor.hpp"
+
+namespace oneapi {
+namespace mkl {
+namespace dft {
+namespace detail {
+
```

Comment:
what's the merit of this ?

Reply:
We don't need to know anything about `commit_impl`, since we're only using pointers to it here. My understanding it that it generally good not to include the full header as a result since we don't need it. Including the full header slows compilation down and also can also create circular include problems.

Reply:
the header will be included in the translation unit anyways once, so I don't understand the need to do this, I don't understand how this might create circular dependency. I would prefer to leave it simple and as is

Reply:
OK, thats leave this as is.